src/main/java/hudson/plugins/sshslaves/verifiers/ManuallyProvidedKeyVerificationStrategy.java

```diff
@@ -92,8 +94,16 @@ private static HostKey parseKey(String key) {
                 RSASHA1Verify.decodeSSHRSAPublicKey(keyValue);
             } else if ("ssh-dss".equals(algorithm)) {
                 DSASHA1Verify.decodeSSHDSAPublicKey(keyValue);
+            } else if ("ssh-ed25519".equals(algorithm)) {
```

Comment:
This should instead be

            } else if (isClass("ED25519KeyAlgorithm") && "ssh-ed25519".equals(algorithm))** {
                 com.trilead.ssh2.signature.ED25519KeyAlgorithm.decodePublicKey(keyValue);
             } else if (isClass("ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp256".equals(algorithm)) {
                 com.trilead.ssh2.signature.ECDSAKeyAlgorithm.decodePublicKey(keyValue);
             } else if (isClass("ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp384".equals(algorithm)) {
                 com.trilead.ssh2.signature.ECDSAKeyAlgorithm.decodePublicKey(keyValue);
             } else if (isClass("ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp521".equals(algorithm)) {
                 com.trilead.ssh2.signature.ECDSAKeyAlgorithm.decodePublicKey(keyValue);
              } else {




then you will want to have a new function probably under this function that does

public boolean isClass(String className) {
 try {
  Class.forName(className);
  return true;
 } catch (ClassNotFoundException e) {
  return false;
 }
}